src/datasources/transaction-api/transaction-api.service.ts

```diff
@@ -241,6 +241,19 @@ export class TransactionApi implements ITransactionApi {
     }
   }
 
+  async getTransfer(transferId: string): Promise<Transfer> {
+    try {
+      const cacheDir = CacheRouter.getTransferCacheDir(
+        this.chainId,
+        transferId,
+      );
+      const url = `${this.baseUrl}/api/v1/transfer/${transferId}`;
+      return await this.dataSource.get(cacheDir, url);
```

Comment:
I think in this case the await statement is necessary, otherwise the catch block wouldn't be executed if `this.datasource.get` rejects. It's a tricky case when using try-catch blocks (for reference: https://stackoverflow.com/a/42750371)

Reply:
Yes, Héctor is correct. The `catch` would not be executed if the promise is rejected that way.

We would need to change the try/catch to chain the success and error cases with the promise itself.